src/micro_http/src/common/headers.rs

```diff
@@ -90,6 +94,8 @@ pub struct Headers {
     /// `Accept` header might be used by HTTP clients to enforce server responses with content
     /// formatted in a specific way.
     accept: MediaType,
+    accept_encoding_flag: bool,
+    accept_encoding: Encoding,
```

Comment:
Instead of having a `field: T` and a `field_flag: bool`, you can make use of Rust's `Option<T>`, but in this particular case the `accept_encoding` field won't be necessary anyway.